pkg/kn/traffic/compute.go

```diff
@@ -0,0 +1,336 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traffic
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/knative/client/pkg/kn/commands/flags"
+	"github.com/knative/pkg/ptr"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/spf13/cobra"
+)
+
+var latestRevisionRef = "@latest"
+
+type ServiceTraffic []v1alpha1.TrafficTarget
+
+func newServiceTraffic(traffic []v1alpha1.TrafficTarget) ServiceTraffic {
+	return ServiceTraffic(traffic)
+}
+
+func splitByEqualSign(pair string) (string, string, error) {
+	parts := strings.SplitN(pair, "=", 2)
+	if len(parts) != 2 {
+		return "", "", errors.New(fmt.Sprintf("expecting the value format in value1=value2, given %s", pair))
+	}
+	return parts[0], strings.TrimSuffix(parts[1], "%"), nil
```

Comment:
Do we still need `strings.TrimSuffix(parts[1], "%")` here, it is harmless but as we don't support `--traffic revision=2%`, `--traffic revision=2` is enough.

Reply:
Right, but we can support both with TrimSuffix, the example section should cover that.